### PR TITLE
Use implicit initializer if present to generate type properties

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/58495

### CODE REVIEW

File: src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs [hunk 1]
```diff
@@ -400,49 +371,54 @@ fieldDeclaration.Parent is CompilationUnitSyntax &&
...
+            AddPropertiesToInitialize(generateTypeServiceStateOptions, initializer);
+        }
 
-                    generateTypeServiceStateOptions.PropertiesToGenerate.Add(name);
-                }
-            }
+        // Check for `NewType t = new() { ... }` and initialize properties if present 
+        if (nameOrMemberAccessExpression.Parent is VariableDeclarationSyntax
+            {
+                Variables: [{ Initializer.Value: ImplicitObjectCreationExpressionSyntax { Initializer: { } implicitInitializer } }, ..]
```
Comment (ToddGrun) [inline src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs:381]: >Variables [](http://example.com/codeflow?start=16&length=9)

Dumb question: Is there any reason we choose the first one? Would it make sense to not enable if more than one variable (or if more than one, only enable if they have the same initialized properties)?

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs:381]: Goal here is to not go to down the rabbit hole of potential cases to support.

Comment (ToddGrun) [reply src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs:381]: chatted offline, Cyrus is being lazy :)

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**